src/renderer/components/CaptionProgram.tsx

```diff
@@ -0,0 +1,336 @@
+};
+
+const run = function(getHasStopped : Function) {
+  if (getHasStopped()) {
+    return;
+  }
+  PROGRAM[programCounter](function() {
+    programCounter += 1;
+    if (programCounter >= PROGRAM.length) {
+      programCounter = 0;
+    }
+    run(getHasStopped);
+  })
+};
+
+
+const startText = function(el : HTMLElement, programText : string) {
+  let i = -1;
+  let hasError = false;
+  let hasStopped = false;
+  const getHasStopped = () => { return hasStopped; };
+
+  programText.split('\n').forEach(function(line) {
+    line = line.trim();
+    i += 1;
+    if (line.length < 1) return;
+
+    if (line[0] === '#') {
+      return;
+    }
+    const command = getFirstWord(line);
+    if (command) {
+      const value = getRest(line);
+      if (COMMANDS[command]) {
+        const fn = COMMANDS[command](el, value);
+        if (fn) {
+          if (command.toLowerCase().startsWith("set")) {
+            fn(getHasStopped);
+          } else {
+            PROGRAM.push(fn);
+          }
+        } else {
+          hasError = true;
+          console.error("Error on line", i, "- invalid arguments");
+        }
+      } else {
+        hasError = true;
+        console.error("Error on line", i, "- unknown command");
+      }
+    }
+  });
+
+  if (!hasError) {
+    run(getHasStopped);
+  }
+  return () => { hasStopped = true; };
+};
+
+
+const startShowingText = function(el : HTMLElement, hastebinId : string) {
+  if (hastebinId === 'test') {
+    let testProgram = `
+    setBlinkDuration 300
+    setBlinkDelay 100
+    setBlinkGroupDelay 1200
+    setCaptionDuration 2000
+    setCaptionDelay 1200
+
+    bigcap YOU LOVE FLUFFY KITTENS
+    blink KITTENS / ARE / YOUR / LIFE
+    cap Cuddle all the kittens forever because you love them.
+    `;
+    return startText(el, testProgram);
+  }
+
+  // const url = `text/${pastebinId}.txt`;
+  // const url = 'https://crossorigin.me/http://pastebin.com/raw/' + pastebinId;
+  // const url = 'http://cors-proxy.htmldriven.com/?url=http://pastebin.com/raw/' + hastebinId;
+  // const url = 'http://cors-proxy.htmldriven.com/?url=https://hastebin.com/raw/' + hastebinId;
+  // const url = 'https://cors-anywhere.herokuapp.com/https://hastebin.com/raw/' + hastebinId;
+  const url = 'https://hastebin.com/raw/' + hastebinId;
+
+  let _hasStoppedEarly = false;
+  let _stop = () => {
+    _hasStoppedEarly = true;
+  };
+  const stop = () => { _stop(); };
+  $.ajax({
+    url: url,
+    method: 'get',
+    success: function(data) {
+      if (_hasStoppedEarly) return;
+      try {
+        JSON.parse(data.body);
+        return;
+      } catch (e) {
+        // ok; old format
+      }
+      if (data.body) {
+        data = data.body;
+      }
+      console.log(data);
+      _stop = startText(el, data);
+    },
+    error: function(err) {
+      console.error("Could not load hastebin ID", hastebinId);
+      alert("Could not load your hastebin. This may or may not be temporary.");
+    },
+  });
+  return stop;
+};
```

Comment:
The `$.ajax` call should be replaced with something like [wretch](https://github.com/elbywan/wretch)